Dapper NET40/SqlMapper.cs

```diff
@@ -4872,6 +4885,63 @@ public void SetValue(IDbDataParameter parameter, object value)
         }
     }
 
+    sealed class SqlDataRecordHandler : Dapper.SqlMapper.ITypeHandler
+    {
+        public object Parse(Type destinationType, object value)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void SetValue(IDbDataParameter parameter, object value)
+        {
+            SqlDataRecordListTVPParameter.Set(parameter, value as IEnumerable<Microsoft.SqlServer.Server.SqlDataRecord>, null);
+        }
+    }
+
+    /// <summary>
+    /// Used to pass a IEnumerable<SqlDataRecord> as a SqlDataRecordListTVPParameter
+    /// </summary>
+    sealed partial class SqlDataRecordListTVPParameter : Dapper.SqlMapper.ICustomQueryParameter
+    {
+        private readonly IEnumerable<Microsoft.SqlServer.Server.SqlDataRecord> data;
+        private readonly string typeName;
+        /// <summary>
+        /// Create a new instance of SqlDataRecordListTVPParameter
+        /// </summary>
+        public SqlDataRecordListTVPParameter(IEnumerable<Microsoft.SqlServer.Server.SqlDataRecord> data, string typeName)
+        {
+            this.data = data;
+            this.typeName = typeName;
+        }
+        static readonly Action<System.Data.SqlClient.SqlParameter, string> setTypeName;

```

Comment:
Being able to write 'WHERE foo IN @Params' into your query, and having dapper expand that into a giant list of parameters. With very large parameter lists it's quite slow, and puts a noticeable CPU load on SQL Server parsing it all.